### DISCUSSION TITLE
input field in options

### DISCUSSION THREAD

**Discussion by rokin04** [JedWatson/react-select#4367] [Q&A]
I'm trying to add an input field in options and update the state value on-change of the input value
https://dz4f6.csb.app/

but input field is not editable !!
can someone help?

**Comment by ebonow** [comment]
Greetings @rokin04 ,

Sorry for the delay but I was attempting to find a similar issue I helped resolved before. 

There are some underlying issues with the approach you are taking that you should be aware of.
1) react-select relies on the input being focused and clicking into a new input would potentially close it
2) The behavior you are likely running into is that the click is attempting to focus your menu input, which blurs the react-select input, but the onMouseDown is in the menuList so it returns focus to the react-select input

To get this working the way you would likely want it to would mean that you would need to manually control the `menuIsOpen` prop to true/false and then add some onMouseDown handler on the Option to focus your input

Hope this is helpful. Let me know if you run into any problems with the suggested implementation
